src/klein/test/test_resource.py

```diff
@@ -30,14 +31,14 @@
 
 
 def requestMock(
-    path,
-    method=b"GET",
-    host=b"localhost",
-    port=8080,
-    isSecure=False,
-    body=None,
-    headers=None,
-):
+    path: bytes,
+    method: bytes = b"GET",
+    host: bytes = b"localhost",
+    port: int = 8080,
+    isSecure: bool = False,
+    body: Optional[bytes] = None,
+    headers: Optional[Mapping[bytes, Sequence[bytes]]] = None,
```

Comment:
Now I know that this is a bit crazy as default headers are shared by all calls... but maybe for this mock is good enough.

I don't know if this would help or will only cause more trouble ;)

```suggestion
    headers: Mapping[bytes, Sequence[bytes]] = {},
```

Reply:
The problem here is with using a mutable object.  Since this function doesn't mutate `headers`, it's arguable safe, but linters will complain, for (I think) good reason.

Reply:
Yes. I would say that not using `None` is a good habit , so it's a pitty that you are force to use None just to work this default Python behaviour.

I see that a frozendict PEP was rejected https://www.python.org/dev/peps/pep-0416/

Not sure if using ImmutableDict as default value would do the job https://werkzeug.palletsprojects.com/en/1.0.x/datastructures/#werkzeug.datastructures.ImmutableDict

Reply:
The best solution I've come across is to use attrs' `Factory`.
In any case, I'm looking forward to throwing this function out in favor of using `HTTPRequest` eventually.